### PR TITLE
Added inlay support to the rascal side

### PR DESCRIPTION
**Pull request by DavyLandman**
Fixing the static part discussed in #119

Screenshot of the inlay hints added to the pico example: 
image

### CODE REVIEW

File: rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/Locations.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 import java.io.IOException;
 import java.net.URISyntaxException;
-
+import org.apache.lucene.search.TopDocs;
 import org.eclipse.lsp4j.Location;
```
Comment (jurgenvinju) [inline rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/Locations.java:32]: Is this one unused?

Comment (DavyLandman) [reply rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/Locations.java:32]: Yup, auto import misfire.

**Review by rodinaarssen — APPROVED** [review]

**Review by jurgenvinju — APPROVED** [review]
This is a *great* feature and the code looks good. I think we should post a TODO somewhere in the rascal-language-servers issue list that as soon lsp4j and VScode have more support for this feature that we simplify the code to use that support.

Also good to add as an issue: asynchronous registration of inlay decorators in the `IDEServices` library.

File: rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/Locations.java [hunk 2]
```diff
@@ -106,4 +106,8 @@ public static Position toPosition(int line, int column, LineColumnOffsetMap map,
         return new Position(line, map.translateColumn(line, column, atEnd));
     }
 
+    public static Position toPosition(ISourceLocation loc, ColumnMaps cm) {
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/Locations.java:108]: it's this class, this is where all the translations happen.

File: rascal-lsp/src/main/rascal/util/LanguageServer.rsc [hunk 3]
```diff
@@ -48,6 +48,7 @@ alias Completer        = list[Completion] (Tree /*input*/, str /*prefix*/, int /
 alias Builder          = list[Message] (list[loc] /*sources*/, PathConfig /*pcfg*/);
 alias LensDetector     = rel[loc src, Command lens] (Tree /*input*/);
 alias CommandExecutor  = void (Command /*command*/);
+alias InlayHinter      = rel[loc src, str label, HintKind kind] (Tree /*input*/);
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply rascal-lsp/src/main/rascal/util/LanguageServer.rsc]: you are looking at an outdated revision.

**Comment by jurgenvinju** [comment]
> Error:    demo/lang/pico/LanguageServer.rsc:72:41: Undefined lexical, layout, alias, nonterminal or keyword `Declaration`

this is causing the build to fail currently.

File: rascal-lsp/src/main/rascal/util/LanguageServer.rsc [hunk 4]
```diff
@@ -123,7 +123,15 @@ data Command(str title="")
     = noop()
     ;
 
-data InlayKind
+data InlayHint
+    = hint(
+        loc range, // should point to a cursor position (begin and end column same)
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply rascal-lsp/src/main/rascal/util/LanguageServer.rsc:128]: It also works correclty.

File: rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java [hunk 5]
```diff
@@ -40,7 +40,7 @@
 import java.util.concurrent.ExecutorService;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
-
+import javax.swing.Icon;
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java]: outdated revision.

File: rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/extensions/InlayHint.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package org.rascalmpl.vscode.lsp.extensions;
+
+import java.util.Objects;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
+
+public class InlayHint {
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/extensions/InlayHint.java:35]: lsp4j tracks the lsp standard, once it's in there (see issue #119 for discussion on the status) we'll migrate to that.

File: rascal-lsp/src/main/rascal/util/LanguageServer.rsc [hunk 7]
```diff
@@ -121,5 +123,11 @@ data Command(str title="")
     = noop()
     ;
 
+data InlayKind
```
[the remark this answers is not in the document]
Comment (DavyLandman) [reply rascal-lsp/src/main/rascal/util/LanguageServer.rsc]: outdated revision.

**Pull request merged**